src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  ChangeEvent,
+  ComponentProps,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends ComponentProps<"input"> {
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  onDebounce?: (searchTerm: string) => void
+  debounceDelay?: number
+  onSearchClick?: (searchTerm: string) => void
+  value?: string
+}
+
+export function SearchInput({
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onDebounce,
+  debounceDelay = 300,
+  onSearchClick,
+  value = "",
+  ...inputProps
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState(value)
+  const [hideCancelButton, setHideCancelButton] = useState(true)
+
+  useEffect(() => {
+    const timer = setTimeout(() => onDebounce?.(searchTerm), debounceDelay)
+    return () => {
+      clearTimeout(timer)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm, debounceDelay])
+
+  const handleReset = (e: MouseEvent<HTMLButtonElement>) => {
+    // we don't want the field to blur when a user clears the input
+    e.preventDefault()
+    setSearchTerm("")
+    setHideCancelButton(true)
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget
+    inputProps.onChange?.(e)
+    setSearchTerm(value)
+    if (!value) setHideCancelButton(true)
+    else setHideCancelButton(false)
+  }
+
+  const handleBlur = (e: FocusEvent<HTMLInputElement>) => {
+    inputProps.onBlur?.(e)
+    setHideCancelButton(true)
+  }
+
+  const handleFocus = (e: FocusEvent<HTMLInputElement>) => {
+    inputProps.onFocus?.(e)
+    if (searchTerm?.length) setHideCancelButton(false)
+  }
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={styles.container} data-testid="container">
+        <span data-testid="search_icon" className={styles.searchIcon}>
+          {searchIcon || <span>🔍</span>}
+        </span>
+        <input
+          {...inputProps}
+          type="search"
+          className={styles.input}
+          onBlur={handleBlur}
+          onFocus={handleFocus}
+          value={searchTerm}
+          name={name}
+          placeholder={placeholder}
+          onChange={handleChange}
+        />
+        <button
+          // we are ignoring tabbing since 'esc' is the keyboard accessible means of clearing an input
+          tabIndex={-1}
+          aria-label="click icon to clear search"
+          hidden={hideCancelButton}
+          className={styles.clearIcon}
+          onMouseDown={handleReset}
```

Comment:
I tend to think of these type of actions happening `onMouseUp` instead of `onMouseDown`, if only because it gives the user an escape hatch if they want to cancel their action. Is there a precedent for using `onMouseDown` here?

Reply:
I'll double back and verify, but I'm pretty certain I had to take this route in order prevent the input from blurring when the clear button is pressed. Otherwise, the onClick and preventDefault wasn't working as expected. Definitely open to learning something here. Maybe I'm not understanding the problem fully.

Reply:
Okay, I figured out what's happening here. It looks like the hidden attribute along with other methods of hiding elements, disables pointer-events...which is the desired behavior...but it seems that the boolean hides the element before onClick has a chance to fire which is why I ended up with onMouseDown. I'm going to see if there is a better strategy here with styling where I can still leverage onClick.

Reply:
So in this case, it seems onMouseDown might be the best strategy to ensure that handleReset is fired before the element is hidden. If you think of a better way, let me know!

Reply:
setting `opacity: 0; pointer-events: none`, `visibility: hidden;`, and `display: none; pointer-events: none`; all seem to remove pointer-events before onClick can be fired. Disabling the element has the same effect as well.

Reply:
seems like something that could be refactored later on if we find a solution. `onMouseDown` it is! ✅